recipes/mcstas/meta.yaml

```diff
@@ -0,0 +1,130 @@
+{% set version = "3.4.7" %}
+{% set sha_value = "e6114c0daa24b77dd69c84846da1505ed8c96b0bc46a91049450ccec21322266" %}
+#NB get sha256 with: curl -sL <URL> | openssl sha256
+
+package:
+  name: mcstas-suite
+  version: {{ version }}
+
+source:
+  url: https://github.com/McStasMcXtrace/McCode/archive/mcstas-{{ version }}.tar.gz
+  sha256: {{ sha_value }}
+  folder: src
+
+build:
+  number: 0
+
+outputs:
+
+  - name: mcstas-core
+    script: install-files-core.sh
+    build:  # [win]
+      skip: True  # [win]
+    requirements:
+      build:
+        - cmake
+        - make  # [not win]
+        - {{ compiler('c') }}
+      host:
+        - flex  # [not win]
+        - bison  # [not win]
+        - winflexbison  # [win]
+        - dirent  # [win]
+      run:
+        - python
+        - {{ compiler('c') }}
+        - bash # for mcrun, and maybe others
+        - pyaml # for mcrun
+        - ply # for traces from mcrun
+    test:
+      files:
+        - test_pkg.py
+      commands:
+        - python test_pkg.py core
+      source_files:
+        - src/mcstas-comps/examples/BNL_H8.instr
+        - src/mcstas-comps/examples/Union_manual_example.instr
+
+  - name: mcstas-data
+    build:
+      noarch: generic
+    script: install-files-data.sh  # [not win]
```

Comment:
What's the motivation for the complex script over just copying over the source folder verbatim - i.e. 
```bash
cp -a "${SRCDATADIR}/." "${DESTDATADIR}/"
```

Reply:
Basically most of that script is mostly just to be defensive in case some unexpected files appears in that folder in the upstream project (it happened in the past) which could cause a headache if simply copying blindly. If the upstream project one day acquires some automated safeguards against that, I will be happy to perform the simplification :-)

Reply:
Having complex scripts that aren't well motivated is a recipe for scaring away reviewers.

I still have no idea why you're writing a complex function to only copy 2 levels deep? It's more to review, and if I don't understand why you're doing it I probably won't hit the merge button (if I could). I'd include comments in the script stating why you're doing these very non-standard things.

I also think it would be better to split into an install and test script. The install script would just be a simple directory copy and after installation you can then run `check_no_forbidden_chars` as a test. Making the install script as bog-standard as possible will likely improve your chances of review from someone who can merge this.

Reply:
Ok thanks, that is a good point! I will try next week to move the sanity checking to the test script instead.